xarray/coding/cftime_offsets.py

```diff
@@ -253,6 +342,121 @@ def onOffset(self, date):
+    _default_month = None  # type: ClassVar[int]
+
+    def __init__(self, n=1, normalize=False, month=None):
+        BaseCFTimeOffset.__init__(self, n)
+        self.normalize = normalize
+        if month is None:
+            self.month = self._default_month
+        else:
+            self.month = month
+        if not isinstance(self.month, int):
+            raise TypeError("'self.month' must be an integer value between 1 "
+                            "and 12.  Instead, it was set to a value of "
+                            "{!r}".format(self.month))
+        elif not (1 <= self.month <= 12):
+            raise ValueError("'self.month' must be an integer value between 1 "
+                             "and 12.  Instead, it was set to a value of "
+                             "{!r}".format(self.month))
+
+    def __apply__(self, other):
+        # months_since: find the calendar quarter containing other.month,
+        # e.g. if other.month == 8, the calendar quarter is [Jul, Aug, Sep].
+        # Then find the month in that quarter containing an onOffset date for
+        # self.  `months_since` is the number of months to shift other.month
+        # to get to this on-offset month.
+        months_since = other.month % 3 - self.month % 3
+        qtrs = roll_qtrday(other, self.n, self.month,
+                           day_option=self._day_option, modby=3)
+        months = qtrs * 3 - months_since
+        return _shift_month(other, months, self._day_option)
+
+    def onOffset(self, date):
+        """Check if the given date is in the set of possible dates created
+        using a length-one version of this offset class."""
+        if self.normalize and not _is_normalized(date):
+            return False
+        mod_month = (date.month - self.month) % 3
+        return mod_month == 0 and date.day == self._get_offset_day(date)
+
+    def __sub__(self, other):
+        import cftime
+
+        if isinstance(other, cftime.datetime):
+            raise TypeError('Cannot subtract cftime.datetime from offset.')
+        elif type(other) == type(self) and other.month == self.month:
+            return type(self)(self.n - other.n, month=self.month)
+        else:
+            return NotImplemented
+
+    def __mul__(self, other):
+        return type(self)(n=other * self.n, month=self.month)
+
+    def rule_code(self):
+        return '{}-{}'.format(self._freq, _MONTH_ABBREVIATIONS[self.month])
+
+    def __str__(self):
+        return '<{}: n={}, month={}>'.format(
+            type(self).__name__, self.n, self.month)
+
+
+class QuarterBegin(QuarterOffset):
+    """Default month for QuarterBegin is December
+    DateOffset increments between Quarter dates.
+
+    month = 1 corresponds to dates like 1/31/2007, 4/30/2007, ...
+    month = 2 corresponds to dates like 2/28/2007, 5/31/2007, ...
+    month = 3 corresponds to dates like 3/31/2007, 6/30/2007, ...
+    """
+    # In pandas, _from_name_startingMonth = 1 used when freq='QS'
```

Comment:
Hmm...this is indeed confusing behavior in pandas.  Reluctantly I'm inclined to match it for consistency, so maybe let's go with `_default_month = 3` here?  In the `_FREQUENCIES` dictionary you can then just use `partial(QuarterBegin, month=1)` for the `'QS'` entry, so that it behaves properly in `to_offset`.